enemies.js

```diff
@@ -77,14 +77,178 @@ class Goomba {
     };
 };
 
-class Koopa {
+class KoopaParatroopaRed {
     constructor(game, x, y, facing) {
-        Object.assign(this, { game, x, y, facing });
+        Object.assign(this, {game, x, y, facing});
+        this.velocity = { x: Math.pow(-2, this.facing)*PARAMS.BITWIDTH, y: 0 }; // pixels per second
+        this.spritesheet = ASSET_MANAGER.getAsset("./sprites/enemies.png");
+        this.animations = [];
+        this.animations.push(new Animator(this.spritesheet, 270, 30, 16, 24, 2, 0.2, 14, false, true));
+        this.animations.push(new Animator(this.spritesheet, 90, 30, 16, 24, 2, 0.2, 14, false, true));
+        this.paused = false;
+        this.dead = false;
+        this.timeLapse = 0;
+        this.deadCounter = 0;
+        this.updateBB();
+    }
+    
+    updateBB() {
+        this.BB = new BoundingBox(this.x, this.y + 1, PARAMS.BLOCKWIDTH, (1 + 7/16) * PARAMS.BLOCKWIDTH);
+    };
+
+    update() {
+        this.timeLapse += this.game.clockTick;
+        if (this.dead) {
+            if (this.deadCounter === 0) this.game.addEntity(new Score(this.game, this.x, this.y, 100));
+            this.deadCounter += this.game.clockTick;
+            this.game.addEntity(new Koopa(this.game, this.x, this.y, this.facing, "red"));
+            this.removeFromWorld = true;
+        }
+        console.log(this.timeLapse);
+        if (!this.paused && !this.dead) {
+            if (this.timeLapse <= 2) {
+                this.y -= 1;
+            }
+            if (this.timeLapse > 2) {
+                this.y += 1;
+            }
+            if (this.timeLapse > 4) {
+                this.timeLapse = 0;
+            }
+                this.y += this.game.clockTick * this.velocity.y * PARAMS.SCALE;
+        }
+        this.updateBB();
+
+        var that = this;
```

Comment:
You can avoid this pattern by using an arrow function on line 123. `(entity) => {` vs `function (entity) {`

Arrow functions don't bind `this` so you can still use it safely in them.

Again optional change but would make the code a bit more clean.